core/src/avm2/object/event_object.rs

```diff
@@ -343,45 +347,36 @@ impl<'gc> EventObject<'gc> {
 
 impl<'gc> TObject<'gc> for EventObject<'gc> {
     fn base(&self) -> Ref<ScriptObjectData<'gc>> {
-        Ref::map(self.0.read(), |read| &read.base)
+        self.0.base.borrow()
     }
 
     fn base_mut(&self, mc: &Mutation<'gc>) -> RefMut<ScriptObjectData<'gc>> {
-        RefMut::map(self.0.write(mc), |write| &mut write.base)
+        unlock!(Gc::write(mc, self.0), EventObjectData, base).borrow_mut()
     }
 
     fn as_ptr(&self) -> *const ObjectPtr {
-        self.0.as_ptr() as *const ObjectPtr
+        Gc::as_ptr(self.0) as *const ObjectPtr
     }
 
     fn value_of(&self, _mc: &Mutation<'gc>) -> Result<Value<'gc>, Error<'gc>> {
         Ok(Value::Object((*self).into()))
     }
 
     fn as_event(&self) -> Option<Ref<Event<'gc>>> {
-        Some(Ref::map(self.0.read(), |d| &d.event))
+        Some(self.0.event.borrow())
     }
 
     fn as_event_mut(&self, mc: &Mutation<'gc>) -> Option<RefMut<Event<'gc>>> {
-        Some(RefMut::map(self.0.write(mc), |d| &mut d.event))
+        Some(unlock!(Gc::write(mc, self.0), EventObjectData, event).borrow_mut())
     }
 }
 
 impl<'gc> Debug for EventObject<'gc> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
-        match self.0.try_read() {
-            Ok(obj) => f
-                .debug_struct("EventObject")
-                .field("type", &obj.event.event_type())
-                .field("class", &obj.base.debug_class_name())
-                .field("ptr", &self.0.as_ptr())
-                .finish(),
-            Err(err) => f
-                .debug_struct("EventObject")
-                .field("type", &err)
-                .field("class", &err)
-                .field("ptr", &self.0.as_ptr())
-                .finish(),
-        }
+        f.debug_struct("EventObject")
+            .field("type", &self.0.event.borrow().event_type())
```

Comment:
technically the borrow is still fallible, can we make it a try_borrow like before?